File: 백태균/week02/12943_콜라츠_추측.java
class Solution {
    public int solution(int num) {

        int cnt = 0;
        long n = num;

        if (n == 1) {
            return 0;
        }

        while (n != 1) {
            if (cnt >= 500) {
                return -1;
            }

            if (n % 2 == 0) {
                n /= 2;
            } else {
                n *= 3;
                n += 1;
            }

            cnt++;
        }

        return cnt;
    }
}

File: 김진효/week02/12943_콜라츠_추측.py
 # [성능 요약] 메모리: 9.2 MB 시간: 0.13 ms 

def solution(num):
    answer = -1
    
    if num == 1:
        return 0
    
    cnt = 0
    while cnt < 500:
        if num % 2 == 0:
            num /= 2
        else:
            num = num*3 + 1
        
        cnt += 1
        
        if num == 1:
            answer = cnt
            break
    
    return answer